### ISSUE TITLE
[BUG] Failed to launch devcontainer

### ISSUE THREAD

**Issue by brooksmtownsend** [wasmCloud/wasmCloud#2070]
### Affected project(s)

- [ ] documentation
- [ ] examples
- [ ] wasmCloud host
- [ ] wasmCloud CLI (wash)
- [ ] wasmCloud dashboard UI (washboard)
- [ ] capability providers
- [ ] provider bindgen
- [ ] control interface client
- [ ] other / not sure

### Describe the bug

```console
➜ devcontainer up --workspace-folder ./             
[0 ms] @devcontainers/cli 0.59.1. Node.js v21.6.2. darwin 23.4.0 arm64.
[424 ms] Start: Run: docker-compose -f /Users/brooks/github.com/wasmcloud/wasmcloud/.devcontainer/docker-compose.yml --profile * config
[523 ms] Start: Run: docker-compose -f /Users/brooks/github.com/wasmcloud/wasmcloud/.devcontainer/docker-compose.yml --profile * config
[605 ms] name: devcontainer
services:
  dev:
    build:
      context: /Users/brooks/github.com/wasmcloud/wasmcloud
      dockerfile: .devcontainer/Dockerfile
      args:
        PRELOAD_EXTENSIONS: arrterian.nix-env-selector
        USER_GID: "1000"
        USER_UID: "1000"
    network_mode: bridge
    security_opt:
      - label:disable
    volumes:
      - type: bind
        source: /Users/brooks/github.com/wasmcloud/wasmcloud
        target: /workspace
        bind:
          create_host_path: true
      - type: volume
        source: nix
        target: /nix
        volume: {}
volumes:
  nix:
    name: devcontainer_nix
[607 ms] Start: Run: docker inspect --type image ghcr.io/xtruder/nix-devcontainer:v1
[760 ms] Start: Run: docker-credential-desktop get
[1066 ms] Docker Compose override file for building image:
version: '3'

services:
  dev:
    build:
      dockerfile: /var/folders/rf/kgpfv4t9437cj3h3c77vglrr0000gn/T/devcontainercli/container-features/0.59.1-1715087922071/Dockerfile-with-features
      args:
        - BUILDKIT_INLINE_CACHE=1
        - _DEV_CONTAINERS_BASE_IMAGE=dev_container_auto_added_stage_label

[1066 ms] Start: Run: docker-compose --project-name wasmcloud_devcontainer -f /Users/brooks/github.com/wasmcloud/wasmcloud/.devcontainer/docker-compose.yml -f /var/folders/rf/kgpfv4t9437cj3h3c77vglrr0000gn/T/devcontainercli/docker-compose/docker-compose.devcontainer.build-1715087922074.yml build
[+] Building 0.2s (7/7) FINISHED                                                                       docker:desktop-linux
 => [dev internal] load build definition from Dockerfile-with-features                                                 0.0s
 => => transferring dockerfile: 989B                                                                                   0.0s
 => [dev internal] load metadata for ghcr.io/xtruder/nix-devcontainer:v1                                               0.2s
 => [dev internal] load .dockerignore                                                                                  0.0s
 => => transferring context: 325B                                                                                      0.0s
 => CANCELED [dev dev_container_auto_added_stage_label 1/2] FROM ghcr.io/xtruder/nix-devcontainer:v1@sha256:c42665ec5  0.0s
 => => resolve ghcr.io/xtruder/nix-devcontainer:v1@sha256:c42665ec512c2e5eacc98ac64441d9b79e81797b2cf4ee7a77bee92cdfd  0.0s
 => => sha256:c42665ec512c2e5eacc98ac64441d9b79e81797b2cf4ee7a77bee92cdfd765b5 3.45kB / 3.45kB                         0.0s
 => => sha256:4edab8b4a6566d31f247b3d402a77ded5d53d9fe62a05d02f5add2ebad67df15 7.41kB / 7.41kB                         0.0s
 => [dev internal] load build context                                                                                  0.0s
 => => transferring context: 2B                                                                                        0.0s
 => CACHED [dev dev_container_auto_added_stage_label 2/2] RUN     if [ -z 1000 ] || [ -z 1000 ] || [ -z code ]; then   0.0s
 => ERROR [dev dev_container_auto_added_stage_label 3/2] COPY .devcontainer/etc/nix.conf /etc/nix/nix.conf             0.0s
------
 > [dev dev_container_auto_added_stage_label 3/2] COPY .devcontainer/etc/nix.conf /etc/nix/nix.conf:
------
failed to solve: failed to compute cache key: failed to calculate checksum of ref 85e7c122-e20c-4726-84af-b7439947196d::vlo196ryspu984ta3b6xbq8p2: failed to walk /var/lib/docker/tmp/buildkit-mount2700792546/.devcontainer/etc: lstat /var/lib/docker/tmp/buildkit-mount2700792546/.devcontainer/etc: no such file or directory
Error: Command failed: docker-compose --project-name wasmcloud_devcontainer -f /Users/brooks/github.com/wasmcloud/wasmcloud/.devcontainer/docker-compose.yml -f /var/folders/rf/kgpfv4t9437cj3h3c77vglrr0000gn/T/devcontainercli/docker-compose/docker-compose.devcontainer.build-1715087922074.yml build
    at Km (/Users/brooks/.nvm/versions/node/v21.6.2/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:429:525)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async QtA (/Users/brooks/.nvm/versions/node/v21.6.2/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:429:2497)
    at async utA (/Users/brooks/.nvm/versions/node/v21.6.2/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:409:3486)
    at async KtA (/Users/brooks/.nvm/versions/node/v21.6.2/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:481:3865)
    at async $C (/Users/brooks/.nvm/versions/node/v21.6.2/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:481:4807)
    at async hrA (/Users/brooks/.nvm/versions/node/v21.6.2/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:661:13255)
    at async lrA (/Users/brooks/.nvm/versions/node/v21.6.2/lib/node_modules/@devcontainers/cli/dist/spec-node/devContainersSpecCLI.js:661:12996)
{"outcome":"error","message":"Command failed: docker-compose --project-name wasmcloud_devcontainer -f /Users/brooks/github.com/wasmcloud/wasmcloud/.devcontainer/docker-compose.yml -f /var/folders/rf/kgpfv4t9437cj3h3c77vglrr0000gn/T/devcontainercli/docker-compose/docker-compose.devcontainer.build-1715087922074.yml build","description":"An error occurred building the Docker Compose images."}
```

### Steps to reproduce

1. navigate to root of repo
2. run `devcontainer up --workspace-root .`
3. see error

### Expected behavior

I expected the devcontainer to stand up

### Environment

- OS: [e.g. Linux, MacOS] MacOS
- Shell [e.g. bash, zsh, powershell] zsh
- wasmcloud Version [e.g. 0.81.0] v1.0.1

### Screenshots / Logs / Additional context

```console
➜ docker compose version  
Docker Compose version v2.24.6-desktop.1

➜ docker --version
Docker version 25.0.3, build 4debf41
```

**Comment by rvolosatovs** [comment]
Looking at #2419, looks like the breakage was introduced in https://github.com/wasmCloud/wasmCloud/commit/f8187aba333d423dd49deb7f61858a1ad5da1f23, when `.devcontainer` was added to top-level `.dockerignore`

**Comment by markkovari** [comment]
Hey @brooksmtownsend we merged the fix hopefully. Would you be so kind to validate and close this if works. Appreciate

**Comment by brooksmtownsend** [comment]
Fixed in #2419!
